db/migrate/20210312004640_add_record_to_access_log.rb

```diff
@@ -0,0 +1,20 @@
+class AddRecordToAccessLog < ActiveRecord::Migration[6.0]
+  def up
+    # add the new columns (does this make indices?)
+    add_reference :access_logs, :record, polymorphic: true
+    # set the new columns
+    update "UPDATE access_logs SET record_type = 'Client', record_id = client_id"
+    # make the new columns non-nullable
+    change_column_null :access_logs, :record_id, false
+    change_column_null :access_logs, :record_type, false
+    # make the old column nullable
+    change_column_null :access_logs, :client_id, true
+  end
+
+  def down
+    # remove the old columns
+    remove_reference :access_logs, :record, polymorphic: true
```

Comment:
One option if you want is to delete the AccessLogs that don't foreign-key to a Client. That's sorta weird so let's that not write that code and hope we never need to do anything like that.